conformance/lang/expressions/unary-expression/unary_minus_expression.balt

```diff
@@ -0,0 +1,1078 @@
+    int d =  -(b ?: 13);
+    io:println(c); // @output -12
+    io:println(d); // @output -13
+}
+
+Test-Case: output
+Description: Test checking expressions as unary negative operands.
+Labels: check, checkpanic, DecimalNumber, error, int, unary-minus, union-type
+
+function init() returns error? {
+    int|error a = 12;
+    int c = -check a;
+    int d = -checkpanic a;
+    io:println(c); // @output -12
+    io:println(d); // @output -12
+}
+
+Test-Case: output
+Description: Test static type of result is contextually expected type.
+Labels: DecimalNumber, decimal, float, int, unary-minus
+
+function init() {
+    float a = -12;
+    decimal b = -12;
+    io:println(a); // @output -12.0
+    io:println(b); // @output -12
+}
+
+Test-Case: error
+Description: Test the static type of operands belonging to non-numeric basic types resulting in an error
+            in unary expression.
+Labels: string, unary-minus
+
+function errorFunction(string a) {
+    string _ = -a; // @error - is not allowed with operands of non-numeric basic types
+}
+
+Test-Case: error
+Description: Test for int the static type of the result being the basic type of the operands via invalid assignment
+             with inference, where operands are of numeric types.
+Labels: decimal, int, float, unary-minus, var
+
+function errorFunction(int a) {
+    var a1 = -a;
+    float _ = a1; // @error static type of unary-minus expression with int operands is int
+
+    var a3 = -a;
+    decimal _ = a3; // @error static type of unary-minus expression with int operands is int
+}
+
+Test-Case: error
+Description: Test for float the static type of the result being the basic type of the operands via invalid assignment
+             with inference, where operands are of numeric types.
+Labels: decimal, int, float, unary-minus, var
+
+function errorFunction(float c) {
+    var a1 = -c;
+    int _ = a1; // @error static type of unary-minus expression with float operands is float
+
+    var a3 = -c;
+    decimal _ = a3; // @error static type of unary-minus expression with float operands is float
+}
+
+Test-Case: error
+Description: Test for decimal the static type of the result being the basic type of the operands via invalid assignment
+             with inference, where operands are of numeric types.
+Labels: decimal, int, float, unary-minus, var
+
+function errorFunction(decimal c) {
+    var a1 = -c;
+    int _ = a1; // @error static type of unary-minus expression with decimal operands is decimal
+
+    var a3 = -c;
+    float _ = a3; // @error static type of unary-minus expression with decimal operands is decimal
+}
+
+Test-Case: error
+Description: Test for optional type the static type of the result being the basic type of the operands via invalid assignment.
+Labels: decimal, int, float, optional-type, unary-minus
+
+function errorFunction(int? a, float? b, decimal? c) {
+    int _ = -a; // @error static type of unary-minus expression int? operand is int?
```

Comment:
Can't we simply give the error as,
```suggestion
    int _ = -a; // @error incompatible types: expected 'int', found 'int?'
```

Refer to https://github.com/ballerina-platform/ballerina-spec/pull/941#discussion_r806482794 also. We don't expect to use terms like "static type" in a compiler error message right.

Reply:
Shall we fix other places?